packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -824,6 +836,22 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
             message.contents = JSON.parse(message.contents);
         }
 
+        // validate client sequence number has no gap. If there is gap, check if there were noops
+        // if there were noops, decrement the gap by number of noops and continue
+        if (this.lastClientSequenceNumber !== undefined && this.lastSubmittedClientId !== undefined && this.lastSubmittedClientId === message.clientId) {
```

Comment:
It feels like this.lastClientSequenceNumber !== undefined should be true if other two checks are true. I.e. I'd remove it from IF statement and make it an assert within IF statement body.

Reply:
the assert fails when the first time we enter the IF block, which is when we see the first "op" op, because lastClientSequenceNumber was not set

Reply:
i think we can just initialize the lastClientSequenceNumber to 0, similar to clientSequenceNumberObserved in connectionManager